### DISCUSSION TITLE
Metadata for the Failure event using Assets not showing up in UI

### DISCUSSION THREAD

**Discussion by NicolaiLolansen** [dagster-io/dagster#17715] [Q&A] [answered]
I can't seem to get the metadata to work when an asset is raising the `Failure` event. The same metadata, when supplied to an `Output` shows up fine in the UI, but not when raising the `Failure`. If I add the metadata to the context using `context.add_output_metadata` it still does not work. If I yield the `Output` with the metadata first, and then raise the `Failure` event it works, but then the asset will show up as materialized, which is not what I want.

Any ideas why this is?

It says in the documentation that it should be rendered in the UI, but also it sits under the "OPs" part of the docs, which might give a hint to me doing something that is an anti-pattern.

https://docs.dagster.io/concepts/ops-jobs-graphs/op-events#failures

**Comment by jamiedemaria** [comment, marked as answer]
Hey @NicolaiLolansen I think there are a couple things going on here:
1. metadata attached to a `Failure` object is interpreted as metadata about the failure itself. For example, if you raise a `Failure` if a column in a dataframe has null values, you might put the number of nulls in the column as metadata on the Failure event. Metadata from Failure events is only viewable in the run logs (i believe this is the case for ops/jobs too, and when I tested I wasn't able to find anywhere else in the UI that displayed the failure metadata). 
2. The metadata that's reported via an `Output` object or from `add_output_metadata` is meant to be about the successfully materialized data asset itself. For example, the number of rows in a database table.  The reason `context.add_output_metadata` doesn't work when you also raise a `Failure` is that the `Failure` halts execution of the asset, so there is no `Output` to attach the metadata to. 

We can definitely reconsider this design though. I can see a lot of value from displaying the metadata from a `Failure` event on the asset info page. Can you open an issue or feature request for this?

**Reply by NicolaiLolansen** [reply]
Thank you @jamiedemaria - I have posted a feature request here: https://github.com/dagster-io/dagster/issues/17752
